kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/config/FormatSelection.scala

```diff
@@ -46,19 +47,18 @@ case object FormatSelection {
 
   def fromKcql(
     kcql: Kcql,
-  ): FormatSelection = {
-
-    val readTextModeFn = () => ReadTextMode(schema.readProps(kcql.getProperties.asScala.toMap))
-
-    Option(kcql.getStoredAs)
-      .map(FormatSelection.fromString(_, readTextModeFn))
-      .getOrElse(JsonFormatSelection)
-  }
+  ): Either[Throwable, FormatSelection] =
+    Option(kcql.getStoredAs) match {
+      case Some(storedAs) =>
+        fromString(storedAs, () => ReadTextMode(schema.readProps(kcql.getProperties.asScala.toMap)))
+      case None =>
+        Right(JsonFormatSelection)
+    }
```

Comment:
Is it nicer to avoid the case match by doing something like this?

```suggestion
    Option(kcql.getStoredAs)
      .map(fromString(_, () => ReadTextMode(schema.readProps(kcql.getProperties.asScala.toMap))))
      .getOrElse(JsonFormatSelection)
```

Reply:
The method returns an Either not an option. `fromstring` returns an Either, so it's changing the "container", so that map does not work. 

This would be the alternative to avoid the pattern match, but I don't think it reads faster.

```    Option(kcql.getStoredAs).fold(JsonFormatSelection.asInstanceOf[FormatSelection].asRight[Throwable]) { storedAs =>
      fromString(storedAs, () => ReadTextMode(schema.readProps(kcql.getProperties.asScala.toMap)))
    }
    ```